### DISCUSSION TITLE
Refactor proposal for margins for facet items

### DISCUSSION THREAD

**Discussion by verheyenkoen** [ugent-library/biblio-backend#1393] [Ideas]
Currently almost all facets has a block like this:

```html
<div class="dropdown-menu__body border-bottom">
    {{$last := sub (len .Hits.Facets.wos_type) 1}}
    {{range $i, $f := .Hits.Facets.wos_type}}
    <div class="form-check {{if lt $i $last}} mb-4{{end}}">
        <input class="form-check-input" id="filter-wos-type-{{$i}}" type="checkbox" name="f[wos_type]" value="{{$f.Value}}"{{if $.SearchArgs.HasFilter "wos_type" $f.Value}} checked{{end}}>
        <label class="form-check-label" for="filter-wos-type-{{$i}}">{{$f.Value}} ({{$f.Count}})</label>
    </div>
    {{end}}
</div>
```

For each item except the last one a margin-bottom is added using the `mb-4` class. Therefor the last item is calculated and the `mb-4` class is added dynamically.

I guess this could be refactored by using the classes `d-flex flex-column gap-4` on the parent div instead.

Thoughts?

**Comment by verheyenkoen** [comment]
(I am aware everything will be converted to templ, but I do think tackling these kind of small refactorings beforehand makes that conversion easier)

**Comment by mietcls** [comment]
The gap property is new, good idea to refactor the facets like this. Could you make a proposal for our UI-kit in our prototype?
Github: https://github.com/ugent-library/biblio-ugent-prototype
Facets are used here: https://booktower-ugent-ui.vercel.app/biblio/researcher/publications/overview.html
UI-kit: https://booktower-ugent-ui.vercel.app/styleguide/docs/introduction.html

**Comment by verheyenkoen** [comment]
Added PR in prototype repo: https://github.com/ugent-library/biblio-ugent-prototype/pull/17
